src/automata/util/builders/automata/TMBuilder.ts

```diff
@@ -0,0 +1,107 @@
+import {AutomataBuilder} from "./AutomataBuilder";
+import {TM} from "../../../non-context-free/TM";
+import {IllegalArgument} from "../../../../exceptions/exceptions";
+import { Alphabet } from "../../../../automata"; 
+import { TMState } from "../../../../states/TMState";
+import { Move } from "../../../../types";
+
+
+export class TMBuilder extends AutomataBuilder<TM>{
+    private readonly _tapeAlphabet;
+
+    constructor(alphabetString: string, _tapeAlphabet : string) {
+        super(alphabetString);
+        this._tapeAlphabet = _tapeAlphabet;
+    }
+
+    
+
+    /**
+     * @override
+     * Override AutomataBuilder withEdges function to include functionality for epsilon edges
+     *
+     * With this function, the Builder can chain addition of edges like:
+     * withEdges.from("state1").to("state2").over("aaRbaR")
+     * 
+     * <state1> --[(a->a,R),(b->a,R)]--> <state2>
+     */
+    public get withEdges(): {
+        from: (start: string) => {
+            to: (end: string) => {
+                over: (symbols: string) => TMBuilder
+            },
+            toSelf: () => {
+                over: (symbols: string) => TMBuilder
+            }
+        }
+    } {
+        return {
+            from: (start: string) => ({
+                to: (to: string) => ({
+                    over: (symbols: string) => {
+                        
+                        if (symbols.length === 0) return this;
+                        // break string abRbbR into array ["abR", "bbR"] empty array if symbols is empty
+                        let parts = symbols.match(/.{1,3}/g) || [];
```

Comment:
~~only here i think it could have to be `/.{3}/g`. Should it be possible to have less than 3 symbols? (perhaps addEdge is defaulting to some values actually?~~

okay it will actually default to a thrown exception so that's good